src/WebJobs.Script/Description/DotNet/DotNetCompilationServiceFactory.cs

```diff
@@ -27,23 +27,39 @@ ImmutableArray<ScriptType> ICompilationServiceFactory.SupportedScriptTypes
             }
         }
 
-        private static OptimizationLevel OptimizationLevel
+        private static bool IsLocal => string.IsNullOrEmpty(Environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteInstanceId));
+
+        private static bool IsRemoteDebuggingEnabled => !string.IsNullOrEmpty(Environment.GetEnvironmentVariable(EnvironmentSettingNames.RemoteDebuggingPort));
+
+        internal static OptimizationLevel OptimizationLevel
         {
             get
             {
                 if (_optimizationLevel == null)
                 {
+                    // Get the release mode setting. If set, this will take priority over environment settings.
                     string releaseModeSetting = Environment.GetEnvironmentVariable(DotNetConstants.CompilationReleaseMode);
```

Comment:
Any reason why your CompilationReleaseMode environment setting isn't in EnvironmentSettings with all the others?